drivers/flash/Kconfig

```diff
@@ -58,6 +58,22 @@ config FLASH_PAGE_LAYOUT
 	help
 	  Enables API for retrieving the layout of flash memory pages.
 
+config FLASH_PARTITION_DEVICE
+	bool "Enable flash partition devices"
+	default n
+	help
+	  The option enables flash partition devices that allow to define
+	  partitions over real flash devices.  This devices use the same
+	  flash API as real devices and can be used in their place.
+
+config FLASH_PARTITION_GENERATE_DEVICE_AT_STARTUP
```

Comment:
Why is this a separate option? Or under what conditions would you want `FLASH_PARTITION_DEVICE` but not `FLASH_PARTITION_GENERATE_DEVICE_AT_STARTUP`?

Reply:
OK, so when I have been posting this I have already been working on macros that allowed to define partitions at run-time: what they would do is to give you flash device that can work with flash API, but will not be visible by `device_get_binding`. When I have been "patch adding" the changes to RFC, this slipped through.